OrbitGl/AsyncTrack.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "AsyncTrack.h"
+
+#include "App.h"
+#include "FunctionUtils.h"
+#include "GlCanvas.h"
+#include "ManualInstrumentationManager.h"
+#include "TimeGraph.h"
+#include "capture_data.pb.h"
+
+using orbit_client_protos::FunctionInfo;
+using orbit_client_protos::TimerInfo;
+
+AsyncTrack::AsyncTrack(TimeGraph* time_graph, const std::string& name) : TimerTrack(time_graph) {
+  SetName(name);
+  SetLabel(name);
+}
+
+[[nodiscard]] std::string AsyncTrack::GetBoxTooltip(PickingId id) const {
+  const TextBox* text_box = time_graph_->GetBatcher().GetTextBox(id);
+  if (text_box == nullptr) return "";
+  ManualInstrumentationManager* manager = time_graph_->GetManualInstrumentationManager();
+  orbit_api::Event event = manager->ApiEventFromTimerInfo(text_box->GetTimerInfo());
+
+  const FunctionInfo* func =
+      GOrbitApp->GetCaptureData().GetSelectedFunction(text_box->GetTimerInfo().function_address());
+
+  std::string function_name = manager->GetString(event.id);
+
+  return absl::StrFormat(
+      "<b>%s</b><br/>"
+      "<i>Timing measured through manual instrumentation</i>"
+      "<br/><br/>"
+      "<b>Module:</b> %s<br/>"
+      "<b>Time:</b> %s",
+      function_name, FunctionUtils::GetLoadedModuleName(*func),
+      GetPrettyTime(
+          TicksToDuration(text_box->GetTimerInfo().start(), text_box->GetTimerInfo().end())));
+}
+
+void AsyncTrack::UpdateBoxHeight() {
+  box_height_ = time_graph_->GetLayout().GetTextBoxHeight();
+  if (collapse_toggle_->IsCollapsed() && depth_ > 0) {
+    box_height_ /= static_cast<float>(depth_);
+  }
+}
+
+void AsyncTrack::OnTimer(const orbit_client_protos::TimerInfo& timer_info) {
+  // Find the first row that that can receive the new timeslice with no overlap.
+  // If none of the existing rows works, add a new row.
+  uint32_t depth = 0;
+  while (max_span_time_by_depth_[depth] >= timer_info.start()) ++depth;
```

Comment:
(I guess these are the two lines one needs to understand why max_span_time_by_depth_ is needed.)

Reply:
I still added a comment in the header file.